lib/SPIRV/SPIRVReader.cpp

```diff
@@ -829,15 +861,8 @@ BinaryOperator *SPIRVToLLVM::transShiftLogicalBitwiseInst(SPIRVValue *BV,
   auto Inst = BinaryOperator::Create(BO, transValue(BBN->getOperand(0), F, BB),
                                      transValue(BBN->getOperand(1), F, BB),
                                      BV->getName(), BB);
-
-  if (BV->hasDecorate(DecorationNoSignedWrap)) {
-    Inst->setHasNoSignedWrap(true);
-  }
-
-  if (BV->hasDecorate(DecorationNoUnsignedWrap)) {
-    Inst->setHasNoUnsignedWrap(true);
-  }
-
+  applyNoIntegerWrapDecorations(BV, Inst);
+  applyFPFastMathModeDecorations(BV, Inst);
```

Comment:
I guess this line can be removed, since we cannot perform logical bitwise shift on floating-point types, can we?

Reply:
No, as this is where translation of binary floating-point arithmetic instructions is handled.

Perhaps the functions should be renamed, and perhaps isBinaryShiftLogicalBitwiseOpCode and isLogicalOpCode should be rewritten to simply list all of the applicable opcodes, in place of the confusing ranges that they currently use.

It would also make sense to split transValueWithoutDecoration() up.  It's 644 lines long.  Something 10-20 lines long that calls helper functions would be better.

Reply:
Ok, thanks for explanation